backend/src/graphql/resolvers/inviteCodes.spec.ts

```diff
@@ -1,214 +1,1134 @@
+          },
+        },
+        errors: undefined,
+      })
+    })
+
+    it('returns a new group invite code with expireDate', async () => {
+      const date = new Date()
+      date.setFullYear(date.getFullYear() + 1)
+      await expect(
+        mutate({
+          mutation: generateGroupInviteCode,
+          variables: { groupId: publicGroup, expiresAt: date.toISOString() },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          generateGroupInviteCode: {
+            code: expect.any(String),
+            comment: null,
+            createdAt: expect.any(String),
+            expiresAt: date.toISOString(),
+            generatedBy: {
+              avatar: {
+                url: expect.any(String),
+              },
+              id: 'inviting-user',
+              name: 'Inviting User',
+            },
+            invitedTo: {
+              id: publicGroup,
+              groupType: 'public',
+              name: 'Public Group',
+              about: 'We are public',
+              avatar: null,
+            },
+            isValid: true,
+            redeemedBy: [],
+          },
+        },
+        errors: undefined,
+      })
+    })
+
+    it('returns a new invalid group invite code with expireDate in the past', async () => {
+      const date = new Date()
+      date.setFullYear(date.getFullYear() - 1)
+      await expect(
+        mutate({
+          mutation: generateGroupInviteCode,
+          variables: { groupId: publicGroup, expiresAt: date.toISOString() },
+        }),
+      ).resolves.toMatchObject({
+        data: {
+          generateGroupInviteCode: {
+            code: expect.any(String),
+            comment: null,
+            createdAt: expect.any(String),
+            expiresAt: date.toISOString(),
+            generatedBy: {
+              avatar: {
+                url: expect.any(String),
+              },
+              id: 'inviting-user',
+              name: 'Inviting User',
+            },
+            invitedTo: {
+              id: publicGroup,
+              groupType: 'public',
+              name: 'Public Group',
+              about: 'We are public',
+              avatar: null,
+            },
+            isValid: false,
+            redeemedBy: [],
+          },
+        },
+        errors: undefined,
+      })
+    })
+  })
+
+  describe('as not-member user', () => {
+    beforeEach(async () => {
+      authenticatedUser = await notMemberUser.toJson()
+    })
+
+    it('throws authorization error', async () => {
+      const date = new Date()
+      date.setFullYear(date.getFullYear() - 1)
+      await expect(
+        mutate({
+          mutation: generateGroupInviteCode,
+          variables: { groupId: publicGroup },
+        }),
+      ).resolves.toMatchObject({
+        data: null,
+        errors: [{ message: 'Not Authorized!' }],
+      })
+    })
+  })
+
+  // eslint-disable-next-line jest/no-disabled-tests
+  describe.skip('as pending-member user', () => {
```

Comment:
TODO

Reply:
I would like to skip this since I cant get a user to be a pending member of a group